src/app/core/guards/passenger.guard.ts

```diff
@@ -5,15 +5,33 @@ import {
   RouterStateSnapshot,
   UrlTree,
   Router,
+  CanLoad,
+  Route,
+  UrlSegment,
 } from '@angular/router';
 import { Observable } from 'rxjs';
 import { AuthService } from '../service/auth.service';
 
 @Injectable({
   providedIn: 'root',
 })
-export class PassengerGuard implements CanActivate {
+export class PassengerGuard implements CanActivate, CanLoad {
   constructor(private router: Router, private authService: AuthService) {}
+  canLoad(
+    route: Route,
+    segments: UrlSegment[]
+  ):
+    | boolean
+    | UrlTree
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree> {
+    if (this.authService.getUserType() === 'Passenger') {
+      return true;
+    } else {
+      this.router.navigate(['/unauthorized']);
+      return false;
+    }
```

Comment:
[ditto](https://github.com/aka-wtag/airline-ticketing-system-frontend/pull/4/files#r1634085917)

Reply:
@ban-wtag apu resolved it.